### DISCUSSION TITLE
How to stop a background script programmatically?

### DISCUSSION THREAD

**Discussion by arthurdenner** [johnlindquist/kit#781] [Q&A]
I have a script with background: true. Under certain conditions, I want to show an error message and stop it.
I'm doing something like this:

```tsx
// Background: true

import '@johnlindquist/kit';
import Stream from 'stream';

let command = `...`;
let writeableStream = new Stream.Writable();

writeableStream._write = async (chunk, __, next) => {
  if (conditionIsTrue) {
    await div(md('### Error message. Try again.'));
    writeableStream.destroy();
    process.exit(1);
  }

  next();
};

exec(command, { all: true }).all.pipe(writeableStream);
```

I get a header with the error message with the above code.
But when I search the script to try again, it shows as running and I need to stop before I can start it again.

Is it possible to stop the script from running in the background programmatically?
If there is a similar discussion or issue, I apologize. I didn't find any.

**Comment by johnlindquist** [comment]
I was just testing it out, it looks like the UI that shows if a `// Background` script is running doesn't update when you terminate the process manually. So you're successfully terminating it, the UI just isn't up-to-date. It's fine to use the UI to "stop" (even though it isn't running) and then start it again. I'll fix it in the next release so that the background red/green indicator stays up-to-date.

One thing to notice is the UI that displays running processes (you can click on this button or use `cmd+p` to open it). That list acts like an "Activity Monitor" and stays up-to-date each second. If your process isn't in that list, it's not running.


![CleanShot 2022-06-14 at 12 14 45@2x](https://user-images.githubusercontent.com/36073/173660432-24177eb7-182c-49d6-b15d-91efaab8c1c5.png)

**Reply by arthurdenner** [reply]
Thanks for the reply. Good to know I'm terminating properly and that the UI will be fixed. Feel free to mark this one as resolved or I can do it when the next release is out.

**Reply by johnlindquist** [reply]
We can wait until the next release. It was a straight-forward fix and I'll try to push out the new release in the next couple of days.

Also, we're starting up a Discord server if you ever want to live-chat script ideas/features/bugs. Happy to hear your input: https://discord.gg/ct8rTSmx
